SecurityPkg/Tcg/Tcg2Dxe/Tcg2Dxe.c

```diff
@@ -2497,14 +2501,32 @@ MeasureSecureBootPolicy (
   IN VOID       *Context
   )
 {
-  EFI_STATUS  Status;
-  VOID        *Protocol;
+  EFI_STATUS    Status;
+  VOID          *Protocol;
+  DEVICE_STATE  CurrentDeviceState; // MU_CHANGE - Measure Firmware Debugger Enabled
 
   Status = gBS->LocateProtocol (&gEfiVariableWriteArchProtocolGuid, NULL, (VOID **)&Protocol);
   if (EFI_ERROR (Status)) {
     return;
   }
 
+  // MU_CHANGE [BEGIN] - Measure Firmware Debugger Enabled
+  CurrentDeviceState = GetDeviceState ();
+
+  if ((CurrentDeviceState & DEVICE_STATE_SOURCE_DEBUG_ENABLED) != 0) {
+    Status = MeasureLaunchOfFirmwareDebugger ();
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, "Failed to measure Firmware Debugger Enabled!\n"));
+      ASSERT_EFI_ERROR (Status);

```

Comment:
My thought was, if we are in debug mode, assert, otherwise if we're in release mode, we want to panic. 

Are you saying we should panic and not assert at all in both scenarios?

Reply:
From a debug point of view, assert vs panic don't make a difference that I'm aware of. You'll stop execution with a message about where you failed. The release build is where the difference is. It's possible the breakpoint vs deadloop functionality of asserts isn't present in panic, but I don't think we need to do both.